api/v3/Case.php

```diff
@@ -240,6 +240,22 @@ function civicrm_api3_case_get($params) {
     $sql->where("a.id IN (SELECT case_id FROM civicrm_case_contact WHERE $clause)");
   }
 
+  // Order by case contact (primary client)
+  if (!empty($options['sort']) && strpos($options['sort'], 'contact_id') === 0) {
```

Comment:
The rest of the parse logic seems thoughtful about edge-cases (like subordinate properties and directionality), so it's surprising that this only kicks in for the first sort-field. (If anything comes before `contact_id.*`, or if there are multiple variants of `contact_id.*`, then they won't work.)

Maybe there's a good reason for this limitation? If it's hard/unnecessary to support multiple `contact_id.*` fields, then maybe throw a more pointed error:

```php
  if (!preg_match(';^contact_id(\.[a-z]+)$;', $options['sort']))) 
    throw Exception("Can only support on one contact_id property");
```